IDE/Renesas/e2studio/RA6M4/test/src/test_main.c

```diff
@@ -0,0 +1,208 @@
+extern "C" {
+void abort(void);
+}
+#endif
+
+
+
+#if defined(TLS_CLIENT) || defined(TLS_SERVER) || defined(EXTRA_SCE_TSIP_TEST)
+
+extern const st_user_key_block_data_t g_key_block_data;
+
+/* Key type of the encrypted user_public_key 0: RSA-2048 2: ECDSA-P256*/
+uint32_t              encrypted_user_key_type = 0;
+
+static int SetScetlsKey()
+{
+#if defined(WOLFSSL_RENESAS_SCEPROTECT)
+
+    #if defined(TLS_CLIENT) || defined(EXTRA_SCE_TSIP_TEST)
+
+      #if defined(USE_CERT_BUFFERS_256)
+        sce_inform_cert_sign((const byte *)ca_ecc_cert_der_sign);
+        encrypted_user_key_type = 2;
+      #else
+        sce_inform_cert_sign((const byte *)ca_cert_der_sign);
+      #endif
+        sce_inform_user_keys(
+            (byte*)&g_key_block_data.encrypted_provisioning_key,
+            (byte*)&g_key_block_data.iv,
+            (byte*)&g_key_block_data.encrypted_user_rsa2048_ne_key,
+            encrypted_user_key_type);
+        #if defined(WOLFSSL_RENESAS_SCEPROTECT_ECC)
+            guser_PKCbInfo.user_key_id = 0; /* not use user key id */
+        #endif
+
+    #elif defined(TLS_SERVER)
+
+        sce_inform_cert_sign((const byte *)client_cert_der_sign);
+        sce_inform_user_keys(
+            (byte*)&g_key_block_data.encrypted_provisioning_key,
+            (byte*)&g_key_block_data.iv,
+            (byte*)&g_key_block_data.encrypted_user_rsa2048_ne_key,
+            encrypted_user_key_type);
+
+    #endif
+
+#endif    
+    return 0;
+}    
+#endif
+
+typedef struct func_args {
+    int    argc;
+    char** argv;
+    int    return_code;
+} func_args;
+
+
+void wolfcrypt_test(func_args args);
+int  benchmark_test(void *args);
+
+
+void sce_test(void)
+{
+    /*(void)timeTick;*/
+
+#if defined(CRYPT_TEST) || defined(BENCHMARK)
+#if defined(CRYPT_TEST)
+    int ret;
+    func_args args = { 0 };
+
+    if ((ret = wolfCrypt_Init()) != 0) {
+         printf("wolfCrypt_Init failed %d\n", ret);
+    }
+
+    printf("Start wolfCrypt Test\n");
+    wolfcrypt_test(args);
+    printf("End wolfCrypt Test\n");
+
+    if ((ret = wolfCrypt_Cleanup()) != 0) {
+        printf("wolfCrypt_Cleanup failed %d\n", ret);
+    }
+#endif
+#if defined(BENCHMARK)
+    #include "hal_data.h"
+    #include "r_sce.h"
+
+    printf("Prepare Installed key\n");
+#if defined(WOLFSSL_RENESAS_SCEPROTECT) && defined(SCEKEY_INSTALLED)
+    /* aes 256 */
+    memcpy(guser_PKCbInfo.sce_wrapped_key_aes256.value,
+           (uint32_t *)DIRECT_KEY_ADDRESS, HW_SCE_AES256_KEY_INDEX_WORD_SIZE*4);
+    guser_PKCbInfo.sce_wrapped_key_aes256.type = SCE_KEY_INDEX_TYPE_AES256;
+    guser_PKCbInfo.aes256_installedkey_set = 1;
+    /* aes 128 */
+    guser_PKCbInfo.aes128_installedkey_set = 0;
+#endif
+    printf("Start wolfCrypt Benchmark\n");
+    benchmark_test(NULL);
+    printf("End wolfCrypt Benchmark\n");
+#endif
+    
+#elif defined(TLS_CLIENT)
+    #include "hal_data.h"
+    #include "r_sce.h"
+    
+  #if defined(USE_CERT_BUFFERS_256)
+   #if defined(TEST_CIPHER_SPECIFIED)
```

Comment:
Added comments to wolfssl_demo.h
```
/* Specifiy cipher suites that are supported by SCE 
 * ClientHello specifies the cipher suite to communicate peer Server 
 * so that TLS handshake uses SCE protect mode 
 */
```